wpical/build.gradle

```diff
@@ -0,0 +1,132 @@
+        wpilibVersionFileOutput.write(read)
+    }
+}
+
+gradle.taskGraph.addTaskExecutionGraphListener { graph ->
+    def willPublish = graph.hasTask(publish)
+    if (willPublish) {
+        generateCppVersion.outputs.upToDateWhen { false }
+    }
+}
+
+def generateTask = createGenerateResourcesTask('main', 'WPIcal', 'wpical', project)
+
+project(':').libraryBuild.dependsOn build
+tasks.withType(CppCompile) {
+    dependsOn generateTask
+    dependsOn generateCppVersion
+}
+
+model {
+    components {
+        // By default, a development executable will be generated. This is to help the case of
+        // testing specific functionality of the library.
+        "${nativeName}"(NativeExecutableSpec) {
+            baseName = 'wpical'
+            sources {
+                cpp {
+                    source {
+                        srcDirs 'src/main/native/cpp', 'src/main/native/thirdparty/mrcal/src', 'src/main/native/thirdparty/mrcal_java/src', "$buildDir/generated/main/cpp"
+                        include '**/*.cpp'
+                    }
+                    exportedHeaders {
+                        srcDirs 'src/main/native/include', 'src/main/native/thirdparty/mrcal/include', 'src/main/native/thirdparty/mrcal_java/include'
+                    }
+                }
+                c {
+                    source {
+                        srcDirs 'src/main/native/thirdparty/libdogleg/src', 'src/main/native/thirdparty/mrcal/src'
+                        include '**/*.c'
+                    }
+                    exportedHeaders {
+                        srcDirs 'src/main/native/include', 'src/main/native/thirdparty/mrcal/generated', 'src/main/native/thirdparty/mrcal/include', 'src/main/native/thirdparty/mrcal/include/minimath', 'src/main/native/thirdparty/libdogleg/include', 'src/main/native/thirdparty/suitesparse_cholmod/include'
+                    }
+                }
+                if (OperatingSystem.current().isWindows()) {
+                    rc.source {
+                        srcDirs 'src/main/native/win'
+                        include '*.rc'
+                    }
+                }
+            }
+            binaries.all {
+                if (it.targetPlatform.name == nativeUtils.wpi.platforms.roborio || it.targetPlatform.name.startsWith("linuxarm")) {
+                    it.buildable = false
+                    return
+                }
+                lib project: ':wpiutil', library: 'wpiutil', linkage: 'static'
+                lib project: ':wpigui', library: 'wpigui', linkage: 'static'
+                lib project: ':thirdparty:imgui_suite', library: 'imgui', linkage: 'static'
+                lib project: ':apriltag', library: 'apriltag', linkage: 'static'
+                nativeUtils.useRequiredLibrary(it, 'ceres')
+                nativeUtils.useRequiredLibrary(it, 'opencv_static')
+                if (it.targetPlatform.operatingSystem.isWindows()) {
```

Comment:
[StackOverflow](https://stackoverflow.com/questions/77912257/error-glog-logging-h-was-not-included-correctly-in-including-glog-in-c-pr) says defining `GLOG_NO_EXPORT` should fix the glog errors, so toss in `it.cppCompiler.define 'GLOG_NO_EXPORT'`.